omreaderlib/src/Interface.scala

```diff
@@ -0,0 +1,61 @@
+package org.chipsalliance.t1.omreaderlib
+
+import scala.reflect.runtime.universe._
+import chisel3.panamalib.option._
+import chisel3.panamaom._
+import chisel3.panamaconverter.PanamaCIRCTConverter
+
+object OMReader {
+  def fromFile(mlirbcFile: os.Path, className: String): OMReader = {
+    new OMReader(os.read.bytes(mlirbcFile), className)
+  }
+
+  def fromBytes(mlirbc: Array[Byte], className: String): OMReader = {
+    new OMReader(mlirbc, className)
+  }
+}
+
+class OMReader private(mlirbc: Array[Byte], className: String) {
+  private val cvt = PanamaCIRCTConverter.newWithMlirBc(mlirbc)
+
+  private val pm = cvt.passManager()
+  assert(pm.populatePreprocessTransforms())
+  assert(pm.populateCHIRRTLToLowFIRRTL())
+  assert(pm.populateLowFIRRTLToHW())
+  assert(pm.populateLowHWToSV())
+  assert(pm.populateExportVerilog(_ => ()))
+  assert(pm.populateFinalizeIR())
+  assert(pm.run())
+
+  private val om = cvt.om()
+  private val evaluator = om.evaluator()
+  private val entry = evaluator.instantiate(className, Seq(om.newBasePathEmpty))
+
+  def entryUnderlying: PanamaCIRCTOMEvaluatorValueObject = entry
+  def t1Reader: T1Reader = new T1Reader(entry)
+}
+
+class T1Reader private[omreaderlib](entry: PanamaCIRCTOMEvaluatorValueObject) {
+  private val om = entry.field("om").asInstanceOf[PanamaCIRCTOMEvaluatorValueObject]
+  private val t1 = om.field("t1").asInstanceOf[PanamaCIRCTOMEvaluatorValueObject]
+
+  def vlen: Long = t1.field("vlen").asInstanceOf[PanamaCIRCTOMEvaluatorValuePrimitiveInteger].integer
+  def dlen: Long = t1.field("dlen").asInstanceOf[PanamaCIRCTOMEvaluatorValuePrimitiveInteger].integer
+
+  def dumpMethods(): Unit = {
+    val mirror = runtimeMirror(getClass.getClassLoader).reflect(this)
+    val methods = typeOf[T1Reader].decls.toList.filter(
+        m => m.isPublic && m.isMethod && !m.isConstructor && !m.asMethod.isGetter
+    )
+    methods.foreach(method => {
+      if (!method.name.toString.startsWith("dump")) {
+        val value = mirror.reflectMethod(method.asMethod)()
+        println(s"${method.name} = $value")
+      }
+    })
+  }
+
+  def dumpAll(): Unit = {
```

Comment:
```suggestion
  def dumpOM(): Unit = {
```

Reply:
what's the purpose of this function?

Reply:
Dump the whole tree of the input instantiated OM object.